WooCommerce/Classes/System/WooNavigationController.swift

```diff
@@ -30,6 +30,19 @@ class WooNavigationController: UINavigationController {
     override var preferredStatusBarStyle: UIStatusBarStyle {
         return StyleManager.statusBarLight
     }
+
+    override func shouldPopOnBackButton() -> Bool {
+        if let vc = topViewController {
+            return vc.shouldPopOnBackButton()
+        }
+        return shouldPopOnBackButton()
+    }
+}
+
+extension WooNavigationController {
+    public override func navigationBar(_ navigationBar: UINavigationBar, shouldPop item: UINavigationItem) -> Bool {
+        self.checkIfNavigationBarShouldPop(navigationBar, item: item)
+    }
```

Comment:
Overriding it in an extension of `WooNavigationController` because when it was just for `UINavigationController` it was not being called.

Reply:
Nice!